### PR TITLE
configure.ac:expand comment for AC_USE_SYSTEM_EXTENSIONS

### PR DESCRIPTION
**Pull request by andy5995**

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -20,7 +20,11 @@ if test -n "${build}" -a -n "${target}"; then
 	GEANY_STATUS_ADD([Building Geany for], [${target}])
 fi
 
-# why do we use this?
+# If possible, enable extensions to C or Posix on hosts that normally
+# disable the extensions, typically due to standards-conformance
+# namespace issues. This should be called before any macros that run the
+# C compiler.
+# <https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Posix-Variants.html>
```
Comment (b4n) [inline configure.ac]: Did you check we actually need it?  If so, do you know for what we do need it?

Comment (andy5995) [reply configure.ac]: Oh, I misunderstood the intent of the commented question.

I couldn't say for sure.. I did a little checking using `git grep` and the only preprocessor macro (from the list shown in that autoconf doc) being used in the Geany repo is '_GNU_SOURCE`.

```
ctags/main/lxcmd.c:46:#ifndef _GNU_SOURCE
ctags/main/lxcmd.c:47:# define _GNU_SOURCE   /* for WIFEXITED and WEXITSTATUS */
ctags/main/options.c:15:#ifndef _GNU_SOURCE
ctags/main/options.c:16:# define _GNU_SOURCE   /* for asprintf */
plugins/export.c:294:#ifdef _GNU_SOURCE
```

Comment (andy5995) [reply configure.ac]: P.S. My guess is that the macro isn't needed, or not being utilized atm. I don't see that _GNU_SOURCE is being defined with a flag during compiling, nor is the `-std=gnu` flag being used.

**Comment by andy5995** [comment]
Ok. All I can say for sure is that when it's removed, Geany still builds. I've attached a diff of the config logs.

[diff.txt](https://github.com/geany/geany/files/2779603/diff.txt)

**Comment by andy5995** [comment]
Stale... closing...

**Pull request closed without merging**